### DISCUSSION TITLE
Error during web setup in K8s

### DISCUSSION THREAD

**Discussion by tmadan** [lowlighter/metrics#972] [Q&A]
Hi

I am trying to set this up in my K8s cluster but running into a problem. Seems like the app is unable to detect the settings.json yaml.

I have mounted the settings.json in /metrics/settings.json and set the env variable SETTINGS as the correct absolute path. Also validated if the file exists but still unable to get it to work.  Also tried to have the file in /tmp but same result.

Error message:

Error: You must provide a valid GitHub personal token to gather your metrics (see https://github.com/lowlighter/metrics/blob/master/.github/readme/partials/documentation/setup/action.md for more informations)
    at file:///metrics/source/app/action/index.mjs:161:13
────────────────────────────────────────────────────────────────────────────────────────
undefined
An error occured, logging debug message :
::error::You must provide a valid GitHub personal token to gather your metrics (see https://github.com/lowlighter/metrics/blob/master/.github/readme/partials/documentation/setup/action.md for more informations)

The token is valid and works when used from my local instance. Any idea how to overcome this issue please? 

Thanks

**Comment by lowlighter** [comment]
Hi 👋 !

In which mode are you trying to run the app 🙂 ?

If you're planning to expose it as a web service on your cluster, indeed you need to mount your `settings.json` to `/metrics/settings.json` but it is also required to override the `entrypoint` to replace it with `npm start` ([see starting docker container as a web service](https://github.com/lowlighter/metrics/blob/master/.github/readme/partials/documentation/setup/web.md#3%EF%B8%8F-start-docker-container)).

However from the logs you report, it seems that it is running in "action" mode, which is the same environment as GitHub Actions runners. 
In this case `settings.json` is actually ignored, since it use environment variables to pass parameters ([see run docker image from cli](https://github.com/lowlighter/metrics/blob/master/.github/readme/partials/documentation/setup/docker.md#1%EF%B8%8F-run-docker-image))

Hope it helps 👍
